operator/bundle_helpers/patch-csv.py

```diff
@@ -85,6 +85,29 @@ def patch_csv(csv_doc, version, operator_image, first_version, no_related_images
         # Ref https://osbs.readthedocs.io/en/latest/users.html?highlight=relatedImages#creating-the-relatedimages-section
         del csv_doc['spec']['relatedImages']
 
+    # Add SecurityPolicy CRD to ACS operator CSV
+    policy_crd = {
+        "name": "securitypolicies.config.stackrox.io",
+        "version": "v1alpha1",
+        "kind": "SecurityPolicy",
+        "displayName": "Security Policy",
+        "description": "SecurityPolicy is the Schema for the policies API",
+        "resources": [{
+            "kind": "Deployment",
+            "name": "",
+            "version": "v1",
+        }],
+    }
+
+    csv_doc["spec"]["customresourcedefinitions"]["owned"].append(policy_crd)
+
+    # Apply policy-as-code feature flag to operator
+    env = csv_doc["spec"]["install"]["spec"]["deployments"][0]["spec"]["template"]["spec"]["containers"][0]["env"]
+    env.append({
```

Comment:
I assume this is currently needed, because the feature flag is off by default -- correct?

I think we should keep `patch-csv.py` as simple as possible, hence I am wondering, if you need a static setting in the CSV, why don't you simply add it to the CSV directly? Am I missing something here?

Reply:
I believe [the CSV gets regenerated from scratch](https://github.com/stackrox/stackrox/blob/35514dd94dd78a62ec7080a084afb383ea06b121/operator/Makefile#L402-L404) during `make bundle`.

Reply:
It gets patched. See https://github.com/stackrox/stackrox/blob/35514dd94dd78a62ec7080a084afb383ea06b121/operator/Makefile#L458

Reply:
Yes, and I am modifying the script that does the patching to add the feature flag.  Perhpas it would be better if I added a more generic feature to the script?  Perhaps adding an `--add-env-var` to allow arbitrary environment variables to be set on the operator's deployment.

Reply:
I am saying, why changing the *script* instead of just modifying the input of the script, namely `bundle/manifests/rhacs-operator.clusterserviceversion.yaml`?
After all you are just adding a static setting.
The script should be only used for dynamic settings, e.g. image tags/digests.
Otherwise we could end up in tricky situtations, because it is not clear which parts of the resulting CSV are coming from the source yaml and which ones are added via the script.

Reply:
`bundle/manifests/rhacs-operator.clusterserviceversion.yaml` is deleted and re-generated from within the `bundle` target:

https://github.com/stackrox/stackrox/blob/1ab0f8534ea26b07d08ddb13b5759a9e2f80fda5/operator/Makefile#L400-L404

I can't see a way to add the static env var to a file without it getting overwritten by the generated output from the `operator-sdk` CLI.

Reply:
I think [adding it here](https://github.com/stackrox/stackrox/blob/ed3a068a7a57052591972c10620529e931eab1d1/operator/config/manager/manager.yaml#L65) might do the trick?

Reply:
But if it's added unconditionally, then why do we need the env variable at all?

Reply:
> I think [adding it here](https://github.com/stackrox/stackrox/blob/ed3a068a7a57052591972c10620529e931eab1d1/operator/config/manager/manager.yaml#L65) might do the trick?

Thank you, that's what I was looking for when I made these review remarks.

Reply:
Thanks for the pointer @porridge!

That said, we just agreed to remove the feature flag altogether in favor of the ability to toggle the feature via Central CR or Helm values file, which means we no longer need to make this change :tada:.

Reply:
In that case let me resolve this.